crates/provider/src/layers/chain.rs

```diff
@@ -0,0 +1,44 @@
+use alloy_chains::NamedChain;
+use alloy_network::Ethereum;
+use alloy_transport::Transport;
+use std::time::Duration;
+
+use crate::{Provider, ProviderLayer};
+
+/// A layer that wraps a [`NamedChain`]. The layer will be used to set
+/// the client's poll interval based on the average block time for this chain.
+///
+/// Does nothing to the client with a local transport.
+#[derive(Debug, Clone, Copy)]
+pub struct ChainLayer(NamedChain);
+
+impl ChainLayer {
+    /// Get the chain's average blocktime, if applicable.
+    pub const fn average_blocktime_hint(&self) -> Option<Duration> {
+        self.0.average_blocktime_hint()
+    }
+}
+
+impl From<NamedChain> for ChainLayer {
+    fn from(chain: NamedChain) -> Self {
+        Self(chain)
+    }
+}
+
+impl<P, T> ProviderLayer<P, T, Ethereum> for ChainLayer
+where
+    P: Provider<T>,
+    T: Transport + Clone,
+{
+    type Provider = P;
+
+    fn layer(&self, inner: P) -> Self::Provider {
```

Comment:
Are there any other properties of a named chain that should affect provider behavior? e.g. do we do any hardfork support tracking in `alloy-chains`?

Reply:
I can only think of `is_legacy()` that should affect eip1599 methods: `eth_maxPriorityFeePerGas`, `eth_feeHistory`

Reply:
could also use `is_legacy` to set some sort of provider or txfiller behavior I suppose. So far we have special-cased 1559 and generally assume it to be active, so I am happy not doing that right now